api/docs/code_reviews.dox

```diff
@@ -233,58 +234,19 @@ to make changes to your code just wastes resources.  Furthermore, in some
-box) to create a clean description of the commit without the messy
-incremental steps from the multiple commits fixing small issues during the
-review process.  (You can resize the edit box by dragging the control in the
-bottom right.)  **Especially be sure to remove *all* commit message titles**
-**from the body of the final squash-and-merge message**, as the commit title is
-in a separate edit box (and defaults to the title of the first commit, so
-update it as well if the title changed during review).  Also be sure to avoid
-truncation of the commit title, as Github likes to replace the end with "...".
-
-To clarify further, here is an example of what Github will present in the
-squash-and-merge step for the message body:
-
-\verbatim
-* i#3192: move delayed branches after markers
-
-Fixes the problem of delayed branches being inserted between the
-timestamp and cpuid marker in drcachesim's offline traces by delaying
-the delayed branch until after all markers.
-
-Fixes: #3192
-
-* Fix clang warning
-
-* Address reviewer comment suggestions
-\endverbatim
-
-All the lines starting with a `*` are commit message titles from commits in the pull request branch.  They should **all** be removed, **including the first one**.  A separate single-line edit box contains the title line for the commit message, which will be added to the front of the description by Github.  Thus, the final description edit box should contain no title at all:
-
-\verbatim
-Fixes the problem of delayed branches being inserted between the
-timestamp and cpuid marker in drcachesim's offline traces by delaying
-the delayed branch until after all markers.
-
-Fixes: #3192
-\endverbatim
-
-When editing the title edit box, leave in place the pointer to the pull request, which Github automatically
-adds to the end of the commit message title, in parentheses, for its suggested final squash-and-merge title:
-
-```
-i#3192: move delayed branches after markers (#3193)
-```
-
-Re-iterating as there is a tendency to just click through the squash-and-merge: _**please edit and**_
-_**clean up the final commit message**_!
+# Merging into Master
+
+Only once the reviewer marks the proposed code changes as accepted, all outstanding
+review comments have been marked resolved, and all of the continuous integration
+tests pass can the change be merged into the master branch.  All commits that are
+part of the pull request branch are squashed together into a single commit on the
+master branch.  The commit message title for this single commit is the pull request
+title, with "(#NNNN)" auto-appended where NNNN is the pull request number.  The
+commit message body is the initial comment at the top level of the pull request: the
+pull request description.  This can be edited after the pull request is created and
+those edits will be used in the squash-and-merge commit message.  The reviewer may
+request changes to the message, which should be performed by editing it through the
+web interface.
```

Comment:
Since we're at it, do we want to add guidelines on: 1) #chars for line-break on the commit message? 2) use of markdown? (it displays nicely on the web interface, but it looks weird looking at plain text merge-commit messages)

Reply:
I added up above in the message content section:

> Use line breaks to keep commit message lines below 80 characters.
>
> Markdown can make descriptions easier to read in the web view, and is allowed.  It
> can make things harder to read in text views through so keep that in mind.